tests/fail/tree-borrows/outside-range.rs

```diff
@@ -0,0 +1,21 @@
+//@compile-flags: -Zmiri-tree-borrows
+fn main() {
+    unsafe {
+        let data = &mut [0u64, 1u64];
+        let raw_data = data.as_mut_ptr();
+        let fst = &mut *raw_data.add(0);
+        let snd = raw_data.add(1);
+        ok(fst);
+        ko(fst, snd);
+        //ko_2(fst, snd as *mut _);
+    }
+}
+
+unsafe fn ok(x: &mut u64) {
+    let _val = *(x as *mut u64).add(1);
+}
+
+unsafe fn ko(x: &mut u64, y: *mut u64) {
+    let _val = *(x as *mut u64).add(1);
+    *y = 100; //~ ERROR: Protector
```

Comment:
This (and other) fail tests also needs a comment at the top explaining what is being tested.